### PR TITLE
chore: ability to disable some plugins via config

### PR DESCRIPTION
**Pull request by Mateuszpietrusinski**
## Description

Giving the ability to select plugins, instead of forcing developers to use our abstraction layers.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

#### Changelog
- [ ] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.

#### Tests
- [ ] I have written test cases for my code
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

- [x] I tested manually my code, and it works well with both:
  - [ ] Default Theme
  - [ ] Capybara Theme

#### Code standards
- [ ] My code follows the code style of this project.
- [ ] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet) 

#### Docs
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by lukasborawski** [comment]
It can't go to the `develop`.

**Review by dawid-ziobro — APPROVED** [review]

File: packages/core/nuxt-module/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  // Add meta data
+  this.options.head.meta.push({
+    name: 'generator',
+    content: 'Vue Storefront 2'
+  });
+
+  log.info('Starting Vue Storefront Nuxt Module');
+
+  // Enable HTTP/2 push for JS files
+  if (options.performance.httpPush) {
```
Comment (lukasborawski) [inline packages/core/nuxt-module/src/index.ts:71]: Isn't this should be acted somehow automatically? On top of some list of files?

Comment (lukasborawski) [reply packages/core/nuxt-module/src/index.ts:71]: Adding a new plugin could be super easy after.

File: packages/core/nuxt-module/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import log from './helpers/log';
+import merge from './helpers/merge';
+import resolveDependency from './helpers/resolveDependency';
+import performanceModule from './modules/performance';
+import storefrontUiModule from './modules/storefront-ui';
+import rawSourcesModule from './modules/raw-sources-loader';
+
+export { default as meta } from '../package.json';
+
+export interface ModuleOptions {
```
Comment (lukasborawski) [inline packages/core/nuxt-module/src/index.ts:13]: Can we move types to the dedicated file?

File: packages/core/nuxt-module/src/helpers/merge.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { mergeWith, isArray } from 'lodash';
```
Comment (lukasborawski) [inline packages/core/nuxt-module/src/helpers/merge.ts:1]: So we've made the ts files but there are no types at all? 🤔

File: packages/core/nuxt-module/src/modules/raw-sources-loader.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import log from '../helpers/log';
+import isProduction from '../helpers/isProduction';
+import resolveDependency from '../helpers/resolveDependency';
+
+export default function VueStorefrontPerformanceModule (options) {
```
Comment (lukasborawski) [inline packages/core/nuxt-module/src/modules/raw-sources-loader.ts:5]: Types are missing.

**Review by lukasborawski — CHANGES_REQUESTED** [review]

**Review by lukasborawski — APPROVED** [review]

File: packages/core/nuxt-module/src/index.ts (hunk 2 above)
Comment (dawid-ziobro) [reply packages/core/nuxt-module/src/index.ts]: Moved

File: packages/core/nuxt-module/src/helpers/merge.ts (hunk 3 above)
Comment (dawid-ziobro) [reply packages/core/nuxt-module/src/helpers/merge.ts:1]: Types added

File: packages/core/nuxt-module/src/modules/raw-sources-loader.ts (hunk 4 above)
Comment (dawid-ziobro) [reply packages/core/nuxt-module/src/modules/raw-sources-loader.ts]: Added

**Pull request merged**